src/third_party_libs/header_only/cereal/external/rapidjson/genericstream.h

```diff
@@ -78,7 +78,7 @@ namespace rapidjson {
       }
 
       size_t Tell() const {
-        return (int)os_.tellp();
+        return static_cast<int>(os_.tellp());
```

Comment:
I didn't want to spend ages on this task so I only replaced the C-style casts where it was obvious what the cast was doing, and that there weren't dozens of them in a given file.  If this was our own source, I'd definitely want them all gone, but since it's Cereal, I thought we'd be OK to leave many in place.

I guess at the end of the day, there are probably hundreds throughout the third-party libs we use, but Cereal (being header-only) gets hit by our compiler flags.

If you want, we could do a Jira task to eliminate all these from Cereal properly and submit a PR to them.  They seem to be pretty keen to be warning-free at a high warning setting.  Until then, if this gets merged, at least our own code is protected from further instances of C-style casts.